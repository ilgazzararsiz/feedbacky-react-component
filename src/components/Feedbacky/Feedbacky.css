.feedbacky__container {
    right: 12px;
    bottom: 12px;
}

.feedbacky__button {
    backface-visibility: hidden;
    border: 1px solid #dcdcdc;
    border-bottom-width: 2px;
    bottom: 5px;
    position: fixed;
    right: 20px;
    width: 40px;
    border: none;
    background-color: transparent;
    color: orange;
    line-height: 42px;
    z-index: 1000;
    position: fixed;
}
