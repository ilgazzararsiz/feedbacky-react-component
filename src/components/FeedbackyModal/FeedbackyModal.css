* {
    box-sizing: border-box;
    margin: 0;
    padding: 0;
}

button {
    background: transparent;
    border: none;
}

.feedbacky__popup {
    display: block;
}

.feedbacky__popup-modal {
    position: fixed;
    width: 100%;
    height: 100%;
    top: 0;
    z-index: 99999;
}

.feedbacky__modal {
    height: 100%;
    width: 100%;
    display: flex;
    justify-content: center;
    align-items: center;
    overflow-y: auto;
    overflow-x: hidden;
    background: rgba(0, 0, 0, 0.2);
}

.feedbacky__modal-content {
    position: relative;
    display: flex;
}

.close__button {
    z-index: 1;
    top: 17px;
    right: 15px;
    position: absolute;
    color: #666;
    cursor: pointer;
    align-self: flex-end;
    border: none;
    width: 30px;
    height: 30px;
}

.close__button-image {
    width: 30px;
    height: 30px;
}

.form__popup {
    float: left;
    background-color: white;
    width: 520px;
    height: 380px;
    margin: auto;
    display: block;
    padding-top: 50px;
}

.form__title {
    text-align: center;
    width: 100%;
    margin-bottom: 10px;
}

.form {
    width: 100%;
    text-align: center;
}

.textarea {
    width: 80%;
    height: 200px;
    resize: none;
}

.send__button {
    backface-visibility: hidden;
    background-color: #f27a1a;
    border: 1px solid #dcdcdc;
    height: 40px;
    line-height: 42px;
    margin-top: 8px;
    margin-bottom: 8px;
    width: 80%;
}

.response__title {
    width: 100%;
    height: 100%;
    text-align: center;
    margin-top: 90px;
    color: green;
}