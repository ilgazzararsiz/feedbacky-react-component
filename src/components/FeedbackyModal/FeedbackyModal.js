import React from 'react'

export const FeedbackyModal = (props) => {
    const handleButtonClose = () => {
        props.handleClose()
    }
    console.log(props)
    return (
        <div className="feedbacky__modal">
            <button onClick={handleButtonClose}>X</button>
            <div>
                <h1>SEND YOUR FEEDBACK</h1>
                <form>
                    <div>
                        <label>
                            <textarea/>
                        </label>
                    </div>
                    <button>SEND</button>
                </form>
            </div>
        </div>
    )
}