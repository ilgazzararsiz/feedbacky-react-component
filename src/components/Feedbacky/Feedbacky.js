import React, {useState} from 'react'
import ChatOutlinedIcon from '@mui/icons-material/ChatOutlined';
import {FeedbackyModal} from '../FeedbackyModal'
import * as PropTypes from "prop-types";


import './Feedbacky.css'

export const Feedbacky = (props) => {
    const [isModalOpen, setModalIsOpen] = useState(false)
    const [isVisible, setIsVisible] = useState(true)

    const handleButtonClick = () => {
        setModalIsOpen(!isModalOpen)
        setIsVisible(!isVisible)
    }

    return (
        <>
            <div className="feedbacky__container">
                {isVisible && (<button className="feedbacky__button" type="button" onClick={handleButtonClick} style={{color: props.buttonColor}}>
                    <ChatOutlinedIcon sx={{fontSize: 40}}/>
                </button>)}
            </div>
            {isModalOpen && <FeedbackyModal handleClose={handleButtonClick} {...props}/>}
        </>
    )
}

Feedbacky.propTypes = {
    applicationId: PropTypes.string,
    buttonColor: PropTypes.string,
    fontColor: PropTypes.string,
    formButtonText: PropTypes.string,
    formButtonColor: PropTypes.string,
    formButtonTextColor: PropTypes.string,
    modalBackgroundColor: PropTypes.string,
    formTitle: PropTypes.string,
    responseTitle: PropTypes.string,
    responseTitleColor: PropTypes.string,
}

Feedbacky.defaultProps = {
    applicationId: "TEST",
    buttonColor: "#f37919",
    fontColor: "#fff",
    formButtonText: "SEND",
    formButtonColor: "#f37919",
    formButtonTextColor: "black",
    modalBackgroundColor: "white",
    formTitle: "SEND YOUR FEEDBACK",
    responseTitle: "WE HAVE GOT YOUR FEEDBACK",
    responseTitleColor: "green"
}