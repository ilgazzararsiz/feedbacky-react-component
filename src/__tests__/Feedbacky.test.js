import React from 'react'
import {render} from "@testing-library/react";
import {Feedbacky} from '../components/Feedbacky'
import {FeedbackyModal} from '../components/FeedbackyModal'

test('Should matches Feedbacky with snapshot', () => {
    const {container} = render(<Feedbacky/>)
    expect(container.firstChild).toMatchSnapshot()
})



