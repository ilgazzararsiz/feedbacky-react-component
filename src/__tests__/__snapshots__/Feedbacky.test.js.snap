// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should matches Feedbacky with snapshot 1`] = `
<div
  class="feedbacky__container"
>
  <button
    type="button"
  >
    <img
      class="feedbacky__button"
      src="feedbacky.png"
    />
  </button>
</div>
`;
